File: services/automation-operations-service/src/topics.rs
//! Topic constants the service publishes to / subscribes from.
//!
//! Re-exports the canonical strings from `libs/saga::events` so the
//! service binary never builds a topic name from a literal — every
//! reference flows through the pinned constant. A drift between
//! the constants and the helm-provisioned topics in
//! `infra/helm/infra/kafka-cluster/values.yaml` is locked in by the
//! `topic_constants_match_helm_provisioning` test below.

pub use saga::events::{
    SAGA_ABORTED_V1, SAGA_COMPENSATE_V1, SAGA_COMPLETED_V1, SAGA_STEP_COMPENSATED_V1,
    SAGA_STEP_COMPLETED_V1, SAGA_STEP_FAILED_V1, SAGA_STEP_REQUESTED_V1,
};

#[cfg(test)]
mod tests {
    use super::*;

    /// Defence in depth — the helm chart provisions these exact
    /// strings (FASE 6 / Tarea 6.2). Keep this test in lockstep with
    /// the kafka-cluster topic catalog.
    #[test]
    fn topic_constants_match_helm_provisioning() {
        assert_eq!(SAGA_STEP_REQUESTED_V1, "saga.step.requested.v1");
        assert_eq!(SAGA_STEP_COMPLETED_V1, "saga.step.completed.v1");
        assert_eq!(SAGA_STEP_FAILED_V1, "saga.step.failed.v1");
        assert_eq!(SAGA_STEP_COMPENSATED_V1, "saga.step.compensated.v1");
        assert_eq!(SAGA_COMPENSATE_V1, "saga.compensate.v1");
        assert_eq!(SAGA_COMPLETED_V1, "saga.completed.v1");
        assert_eq!(SAGA_ABORTED_V1, "saga.aborted.v1");
    }
}

File: services/workflow-automation-service/internal/automationoperations/topics.go
// Package automationoperations holds the saga substrate constants
// (legacy automation-operations-service, S8 merge per ADR-0030).
//
// Kafka topic constants are pinned here — the Rust source re-exports
// from libs/saga::events; the Go port inlines them since libs/saga-go
// is not yet ported. Drift from helm-provisioned values is locked
// by the topic_constants_match_helm_provisioning test.
package automationoperations

// Saga topics (FASE 6 / Tarea 6.2 helm-provisioned).
const (
	SagaStepRequestedV1   = "saga.step.requested.v1"
	SagaStepCompletedV1   = "saga.step.completed.v1"
	SagaStepFailedV1      = "saga.step.failed.v1"
	SagaStepCompensatedV1 = "saga.step.compensated.v1"
	SagaCompensateV1      = "saga.compensate.v1"
	SagaCompletedV1       = "saga.completed.v1"
	SagaAbortedV1         = "saga.aborted.v1"
)

// SagaConsumerGroup is the group id the saga step consumer uses.
// Verbatim from Rust src/automation_operations/domain/saga_consumer.rs.
const SagaConsumerGroup = "automation-operations-service"

// ProcessedEventsTable is the Postgres dedup table for the saga
// consumer's record-before-process rule.
const ProcessedEventsTable = "automation_operations.processed_events"
